sound/soc/sof/intel/hda.c

```diff
@@ -1026,7 +1025,8 @@ static int hda_generic_machine_select(struct snd_sof_dev *sdev)
 
 	/* used by hda machine driver to create dai links */
 	if (pdata->machine) {
-		mach_params = (struct snd_soc_acpi_mach_params *)
+		struct snd_soc_acpi_mach_params *mach_params =
+			(struct snd_soc_acpi_mach_params *)
```

Comment:
I am not a big fan of this, we don't do it anywhere else.

Reply:
+1 I dont see any added value in reducing the scope for the variable in this function too.

Reply:
@plbossart @ranj063 In this specific function moving just one variable inside the block where it is used might not make a huge difference, sure. It's "just" one variable - it's easy enough to check where it's used. But in general you'll probably agree, that it is a good practice to limit the scope of variables to the smallest code block, where they are used. Firstly not to clutter the function scope, i.e. to make it easier to understand the function as a whole, to see which variables are used there. Secondly to easily see where that specific variable is needed. Putting it inside that block immediately tells the reader, that that variable isn't used anywhere outside of it. Both these aspects help readability, I think, and should be encouraged. There are also edge cases like
```
static void f(void)
{
	int i;
	if (x) {
		for (i = 0; i < N; i++) {}
	}
	if (y) {
		for (i = 0; i < N; i++) {}
	}
}
```
In the above case the variable `i` is only needed in respective `if` clauses and it doesn't have to carry its value from one clause to another, so you can define it within each `if` clause. OTOH if the function is simple enough, doesn't have many variables defined in its scope, and if `i` maybe even used more than twice - you might as well save defining it in each enclosing block and just define it once for the whole function.
In short - this isn't something important, but in general, especially where there are many variables, that make the function header difficult to grasp, and if some of those variables clearly are only needed in some limited scope within the function, it's better to have their definitions limited to that scope.

Reply:
I've dropped this one for now, let me know if it's wanted after all

Reply:
@lyakh @plbossart @ranj063 This is interesting. I was under (false) impression that C99 style declarations are not used at least in ALSA, but upon checking the latest codebase, that's not really the case. In ASoC as well, we have C99 style declarations now, e.g. in soc-core.c:snd_soc_add_controls() and snd-pcm.c:soc_new_pcm(). Of course, in SOF we don't have them, so maybe better stick to existing style unless with a very good reason, but at least something to consider. I'm more used to C99 style local declarations myself, but I admit the older kernel coding style of keeping functions short, helps with many of the same issues (i.e. if you have more than 6-7 local variables in your function, you are doing something wrong).

Reply:
@kv2019i sorry, which C99 style declarations do you mean? If you're talking about declaring variables at the block ("compound statement" K&R) level, that's ANSI C, it's also present in K&R. I'm not proposing `for (int i = 0;...)`

Reply:
@lyakh That's good a clarification. C99 relaxed the rules, so I tend to just refer to it when talking about more C++ like style to introduce variables close to place of use, but you are 100% right declarations at start of compounds have been valid from the start. It just has been a bit error prone as gcc has been more relaxed already before C99, so if you used local variable in middle of functions, and gcc, it was easy to introduce gcc-extension'isms to your code. But in kernel, I did not find any examples of real C99 style local variables. That might still be frowned upon.

Reply:
let's stick to traditional kernel style, we have more pressing things to argue about

Reply:
@plbossart I have dropped this patch upon popular request, but I'm pretty sure this isn't working against the traditional kernel style. I don't remember *ever* seeing this kind of code rejected on grounds of style. Of course the kernel consists of many subsystems and each subsystem has its own preferences / traditions, and I still don't know ALSA preferences well enough to argue about them specifically. If this is indeed the preferred style within ALSA or ASoC, please let me know!

Reply:
In the kernel there are two major differing styles with small differences, namely networking and everything else.